util/reggen/gen_dv.py

```diff
@@ -72,7 +104,41 @@ def gen_core_file(outdir: str,
         yaml.dump(core_data, core_file, encoding='utf-8')
 
 
-def gen_dv(block: IpBlock, dv_base_prefix: str, outdir: str) -> int:
+def get_dv_base_names_objects(dv_base_names: List[str]) -> Dict[str, DvBaseNames]:
+    '''Convert the dv_base_names from string to a list of DvBaseNames objects.
+
+    If the `dv_base_names` matches the default argument, returns default DvBaseNames object.
+    If the `dv_bave_names` has customerized input, it should follow this format of a list:
+    ['block-1:type:entity-name', 'block-2:type:entity-name',...].
+    This method will parse the list and return customerized DvBaseNames objects.
+    '''
+    if dv_base_names is None:
+        return None
+
+    dv_base_names_dict = defaultdict(DvBaseNames)  # type: Dict[str, DvBaseNames]
+
+    for item in dv_base_names:
+        try:
+            block, base_type, entity = item.split(":")
+        except ValueError:
+            log.error(f"Bad input arg: {item}")
+            sys.exit(1)
+        dv_base_names_dict[block].set_entity(base_type, entity)
+    return dv_base_names_dict
+
+
+def get_block_base_name(dv_base_names_map: Dict[str, DvBaseNames], block: str) -> DvBaseNames:
+    '''Given a Dictionary of DvBaseNames and return a DvBaseNames object for a specific block.
+    '''
+    try:
+        return dv_base_names_map[block]
+    # If does not find matching key or the map is empty, return the default
+    # DvBaseNames object.
+    except (KeyError, TypeError):
```

Comment:
Seems like this will pass if I do `get_block_base_name(12345, "ast")`. Wonder if the `None` case should be handled separately. Not a big deal though...

Reply:
That is true! I will update the code for that :) thanks for pointing it out.